torch/csrc/jit/register_prim_ops_c10.cpp

```diff
@@ -0,0 +1,36 @@
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/ATen.h>
+#include <ATen/core/stack.h>
+
+using Stack = std::vector<c10::IValue>;
+using torch::jit::peek;
+using torch::jit::drop;
+using torch::jit::pack;
+using torch::jit::push;
+using torch::jit::pop;
+using at::Tensor;
+using at::Scalar;
+using c10::IValue;
+
+static auto registry_prim = torch::RegisterOperators().op("aten::Int.Tensor(Tensor a) -> int",
+  torch::RegisterOperators::options().catchAllKernel(
```

Comment:
`torch::RegisterOperators::options().catchAllKernel` <- this boilerplate is really big. Can we make this shorter? There are a log of registrations in register_prim_ops and having to read around it will be difficuilt.

Reply:
@smessmer Could we make it shorter?

Reply:
cc @charliechen0401 on if we can already skip the `aliasAnalysis()` call here. If we can skip the `aliasAnalysis()` call, then we can use the shorthand API:

    static auto registry_prim = torch::RegisterOperators().op("aten::Int.Tensor(Tensor a) -> int", [] (at::Tensor a) -> int64_t {...}))

Reply:
oh wait those ops aren't specified in `native_functions.yaml`. Then, we actually do need to specify `aliasAnalysis()` and the shorthand doesn't work.

Reply:
Yep, aliasAnalysis() is required here. Otherwise there will be runtime error complaining on the in-consistent input shape.